### PR TITLE
add api catalog test #test(ONEPLAT-1698)

### PR DESCRIPTION
**Pull request by mudit94**
# Closes/Fixes/Resolves

### 

# Explain the feature/fix
Added TC for add api catalog test 

## Does this PR introduce a breaking change
No

## Screenshots

api-cattalog

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did tests pass?
- Yes
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: e2e/integration/apicatalogtests.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        const words = cy.faker.lorem.words();
+        cy.get( 'input[name="name"]' ).type( words );
+        cy.get( 'textarea[name="description"]' ).type( "test" )
+        cy.contains( "REST" ).click( { force: true } )
+        cy.get( 'div[class="pf-c-select__toggle-wrapper"] > input[placeholder^="Search"]' ).type( 'one-portal' ).then( () => {
+            cy.wait(5000)
+            cy.contains( 'one-portal-test', { timeout: 10000 } ).should( 'be.visible' ).click( { force: true } )
+        })
+        cy.get( '[name="appUrl"]' ).type( "https://www.google.com" )
+        cy.get( '[name="schemaEndpoint"]' ).type( "https://www.redhat.com" )
```
Comment (riginoommen) [inline e2e/integration/apicatalogtests.js:25]:
```suggestion
        cy.get( '[name="schemaEndpoint"]' ).type( "https://petstore.swagger.io/v2/swagger.json" )
```

Comment (mudit94) [reply e2e/integration/apicatalogtests.js]: Hi @riginoommen , I have updated this in https://github.com/1-Platform/one-platform/pull/1317/commits/5c5599e2d31a8fab11c05cb778b1fd2c725c6261

File: e2e/integration/apicatalogtests.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        cy.contains( 'Add API' ).click( { force: true } );
+    } )
+    it( 'Fill the add API form', () => {
+        cy.faker = require( 'faker' );
+        const words = cy.faker.lorem.words();
+        cy.get( 'input[name="name"]' ).type( words );
+        cy.get( 'textarea[name="description"]' ).type( "test" )
+        cy.contains( "REST" ).click( { force: true } )
+        cy.get( '#app-layout-page > section.pf-c-page__main-section.pf-m-no-padding.pf-m-light.appLayout_app-layout--content__1Gx8I > section > div > div > div > div:nth-child(2) > form > div:nth-child(4) > div.pf-c-form__group-control > div' ).within( () => {
+            cy.get('input[placeholder ^= "Search" ]' ).type( 'one-portal-test' ).then( () => {
```
Comment (riginoommen) [inline e2e/integration/apicatalogtests.js:27]: Use env variables

**Review by riginoommen — CHANGES_REQUESTED** [review]
Please check the above comments

**Review by ghanlohar — APPROVED** [review]
Apart from @riginoommen's comment, Looks good to me.

File: e2e/integration/apicatalogtests.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            cy.wait( 5000 );
+            cy.contains( 'sso', { timeout: 10000 } ).should( 'not.be.disabled' ).click( { force: true } );
+            // cy.contains( 'one-portal-test sso-tester', { timeout: 10000 } ).click({force:true})
```
[the remark this answers is not in the document]
Comment (mudit94) [reply e2e/integration/apicatalogtests.js:30]: Hi @riginoommen , the change has been done https://github.com/1-Platform/one-platform/pull/1317/commits/2cf70cc56138493cab30aea827164dbe610a37e6

[the remark this answers is not in the document]
Comment (riginoommen) [reply e2e/integration/apicatalogtests.js:30]: Still the commented line is not removed

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: e2e/integration/apicatalogtests.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (hybridx) [reply e2e/integration/apicatalogtests.js:30]: @mudit94 please remove the commented line

**Review by hybridx — APPROVED** [review]
Please remove the comment rest looks good to me

File: e2e/integration/apicatalogtests.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mudit94) [reply e2e/integration/apicatalogtests.js]: @riginoommen , @hybridx , please check https://github.com/1-Platform/one-platform/pull/1317/commits/921a6420568759cec948ebb10a05a585bc9d5a2c

[the remark this answers is not in the document]
Comment (hybridx) [reply e2e/integration/apicatalogtests.js]: Done thanks @mudit94

**Review by riginoommen — APPROVED** [review]

File: e2e/integration/apicatalogtests.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+/// <reference types="Cypress" />
+Cypress.on( 'uncaught:exception', ( err, runnable ) => {
+    // returning false here prevents Cypress from
+    // failing the test
+    return false;
+} )
+context( 'Home Page Tests', () => {
+    before( () => {
+        cy.visit( Cypress.env( 'QA_HOST' ) +'developers/api-catalog');
```
Comment (riginoommen) [inline e2e/integration/apicatalogtests.js:9]: As i see in before PR also Why this env change do we have to update jenkins also?

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: e2e/integration/apicatalogtests.js (hunk 4 above)
Comment (hybridx) [reply e2e/integration/apicatalogtests.js:9]: Nope

**Review by hybridx — CHANGES_REQUESTED** [review]

**Review by riginoommen — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

**Pull request merged**